pkgs/os-specific/linux/msi-ec/patches/Makefile

```
@@ -0,0 +1,21 @@
+# Out of the box, the build with this Makefile only works in FHS environments,
+# such as on Ubuntu or Debian. On NixOS, you either need to open an FHS
+# environment using a Nix shell or build this from a specially crafted Nix
+# derivation.
+#
+# This file follows the conventions written down here:
+# https://docs.kernel.org/kbuild/modules.html
+# Make it possible to override the kernel src tree location from Nix derivation.
+KERNEL ?= $(shell uname -r)
+KERNELDIR ?= /lib/modules/$(KERNEL)/build/
```

```suggestion
KERNELDIR ?= /lib/modules/$(KERNEL)/build/
ccflags-y := -std=gnu11 -Wno-declaration-after-statement
```

These are the [flags used by upstream](https://github.com/m1dugh/msi-ec/blob/e261feba4ae92d1626ed5a9c1ce89883b303524c/Makefile#L5). We should use them if possible.

Docs about this: https://www.kernel.org/doc/Documentation/kbuild/makefiles.txt